core/math/math_defs.h

```diff
@@ -31,6 +31,9 @@
 #ifndef MATH_DEFS_H
 #define MATH_DEFS_H
 
+#define CMP_EPSILON_32 0.00001
+#define CMP_EPSILON_64 0.000000001
```

Comment:
Alternate to the above: If we decide to keep the methods as they are in this PR, with the double methods using an epsilon more precise than what 32-bit floats are expected to have in error, it would still be a nice idea to have stricter epsilon values available when bug hunting.

```suggestion
#ifdef PRECISE_MATH_CHECKS
#define CMP_EPSILON_32 0.000001
#define CMP_EPSILON_64 0.00000000000001
#else
#define CMP_EPSILON_32 0.00001
#define CMP_EPSILON_64 0.000000001
#endif // PRECISE_MATH_CHECKS
```